src/vm/appdomain.cpp

```diff
@@ -7535,22 +7578,40 @@ void AppDomain::CheckForMismatchedNativeImages(AssemblySpec * pSpec, const GUID
         //
         // No entry yet - create one
         //
-        AllocMemTracker amTracker;
-        AllocMemTracker *pamTracker = &amTracker;
+        NativeImageDependenciesEntry * pNewEntry = new NativeImageDependenciesEntry();
```

Comment:
Because we want to do a `RemoveNativeImageDependency` (and a delete of `NativeImageDependenciesEntry`)... not sure how to do this with the loader heap?